contracts/ClearingHouse.sol

```diff
@@ -920,30 +848,32 @@ contract ClearingHouse is
             0
         );
 
-        if (response.pnlToBeRealized != 0) {
-            // if realized pnl is not zero, that means trader is reducing or closing position
-            // trader cannot reduce/close position if the remaining account value is less than
-            // accountValue * LiquidationPenaltyRatio, which
-            // enforces traders to keep LiquidationPenaltyRatio of accountValue to
-            // shore the remaining positions and make sure traders having enough money to pay liquidation penalty.
-
-            // CH_NEMRM : not enough minimum required margin after reducing/closing position
-            require(
-                getAccountValue(params.trader) >=
-                    _getTotalAbsPositionValue(params.trader).mulRatio(_getLiquidationPenaltyRatio()).toInt256(),
-                "CH_NEMRM"
-            );
-        }
-
-        // check margin ratio after swap: mmRatio for closing position; else, imRatio
-        if (params.isClose) {
-            // CH_NEFCM: not enough free collateral by mmRatio
-            require(
-                (_getFreeCollateralByRatio(params.trader, IClearingHouseConfig(_clearingHouseConfig).getMmRatio()) >=
-                    0),
-                "CH_NEFCM"
-            );
+        if (takerPositionSizeBeforeSwap != 0) {
```

Comment:
For this part, I couldn't think of a way to have a good readability + lower gas.